libs/remix-ui/tabs/src/lib/components/CompileDropdown.tsx

```diff
@@ -0,0 +1,222 @@
+import { AppModal } from '@remix-ui/app'
+import { FormattedMessage } from 'react-intl'
+import { SolScanTable, ScanReport } from '@remix-ui/tabs'
+import axios from 'axios'
+import { endpointUrls } from '@remix-endpoints-helper'
+
+import { ArrowRightBig, IpfsLogo, SwarmLogo, SettingsLogo, SolidityScanLogo, AnalysisLogo, TsLogo  } from '@remix-ui/tabs'
+
+const _paq = (window._paq = window._paq || [])
+
+interface CompileDropdownProps {
+  tabPath?: string
+  plugin?: any
+  disabled?: boolean
+  compiledFileName?: string
+  onNotify?: (msg: string) => void
+  onOpen?: () => void
+  onRequestCompileAndPublish?: (type: string) => void;
+}
+
+export const CompileDropdown: React.FC<CompileDropdownProps> = ({ tabPath, plugin, disabled, onNotify, onOpen, onRequestCompileAndPublish, compiledFileName }) => {
+  const [scriptFiles, setScriptFiles] = useState<string[]>([])
+
+  const fetchScripts = async () => {
+    try {
+      let files = {}
+      const exists = await plugin.call('fileManager', 'exists', 'scripts')
+      if (exists) {
+        files = await plugin.call('fileManager', 'readdir', 'scripts')
+      }
+      const tsFiles = Object.keys(files).filter(f => f.endsWith('.ts'))
+      setScriptFiles(tsFiles)
+      onNotify?.(`Loaded ${tsFiles.length} script files`)
+    } catch (err) {
+      console.error("Failed to read scripts directory:", err)
+      onNotify?.("Failed to read scripts directory")
+    }
+  }
+
+  const runScript = async (path: string) => {
+    await plugin.call('solidity', 'compile', tabPath)
+    const content = await plugin.call('fileManager', 'readFile', path)
+    await plugin.call('scriptRunnerBridge', 'execute', content, path)
+    onNotify?.(`Executed script: ${path}`)
+  }
+
+  const runRemixAnalysis = async () => {
+    _paq.push(['trackEvent', 'solidityCompiler', 'staticAnalysis', 'initiate'])
+    await plugin.call('solidity', 'compile', tabPath)
+    const isStaticAnalyzersActive = await plugin.call('manager', 'isActive', 'solidityStaticAnalysis')
+    if (!isStaticAnalyzersActive) {
+      await plugin.call('manager', 'activatePlugin', 'solidityStaticAnalysis')
+    }
+    plugin.call('menuicons', 'select', 'solidityStaticAnalysis')
+    onNotify?.("Ran Remix static analysis")
+  }
+
+  const handleScanContinue = async () => {
+    await plugin.call('solidity', 'compile', tabPath)
+    await plugin.call('notification', 'toast', 'Processing data to scan...')
+    _paq.push(['trackEvent', 'solidityCompiler', 'solidityScan', 'initiateScan'])
+
+    const workspace = await plugin.call('filePanel', 'getCurrentWorkspace')
+    const fileName = `${workspace.name}/${compiledFileName}`
+    const filePath = `.workspaces/${compiledFileName}`
+    const file = await plugin.call('fileManager', 'readFile', filePath)
+
+    const urlResponse = await axios.post(`${endpointUrls.solidityScan}/uploadFile`, { file, fileName })
+
+    if (urlResponse.data.status === 'success') {
```

Comment:
Is this code has been copied over from the solidity tab? If possible it's better to have a helper function for this.

Reply:
Extracted the duplicated Solidity Scan API call and WebSocket logic into a new helper function.